packages/neon/neon_talk/lib/src/widgets/rich_object/file.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:file_icons/file_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:neon_framework/blocs.dart';
+import 'package:neon_framework/theme.dart';
+import 'package:neon_framework/utils.dart';
+import 'package:neon_framework/widgets.dart';
+import 'package:nextcloud/core.dart';
+import 'package:nextcloud/spreed.dart' as spreed;
+
+/// Displays a file preview from a rich object.
+class TalkRichObjectFile extends StatelessWidget {
+  /// Creates a new Talk rich object file.
+  const TalkRichObjectFile({
+    required this.parameter,
+    super.key,
+  });
+
+  /// The parameter to display.
+  final spreed.RichObjectParameter parameter;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget child;
+
+    if (parameter.previewAvailable == spreed.RichObjectParameter_PreviewAvailable.yes) {
+      final maxHeight = MediaQuery.of(context).size.height ~/ 2;
```

Comment:
Yes, I wanted to limit the max height an image can take up on the screen to avoid it filling out the whole screen at once.

Reply:
Do you think there is a better way to do this? The image should not always just take the full original size.

Reply:
I just wanted to mention that `MediaQueryData.size` usually is the screen size of the device and not the widget size (depending on where the nearest MediaQuery ancestor is in the WidgetTree).
If the widget is constrained or other things on the screen take up room like an appBar they are not considered.
I _think_ that a LayoutBuilder is preffered in this scenario.

The algorithm itself is ok. Just the way you obtain the height should change.

Reply:
LayoutBuilder won't work because it is inside a list view so the max height constraints is infinity. I think using the window size is fine (it's not the screen size btw).